config/main.py

```diff
@@ -524,6 +524,25 @@ def load_minigraph():
     click.echo("Please note setting loaded from minigraph will be lost after system reboot. To preserve setting, run `config save`.")
 
 
+#
+# 'change-hostname' command
+#
+@config.command('change-hostname')
+@click.argument('hostname', metavar='<hostname>', required=True)
+def change_hostname(hostname):
+    """Change Hostname on a SONiC device without impacting the traffic."""
+
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    config_db.mod_entry('DEVICE_METADATA' , 'localhost', {"hostname" : hostname})
+    try:
+        command = "service hostname-config restart"
```

Comment:
Do we need to restart `hostname-config` service? Isn't [this ](https://github.com/Azure/sonic-buildimage/blob/master/files/image_config/hostcfgd/hostcfgd#L312)handles the config change?

Reply:
Not sure. 🤔 I had to restart to "hostname-config" to make it effective.

Reply:
Hmm... I believe Prince is correct -- hostcfgd should now pick up the change and apply it without a restart. This change went in ~4 months ago. @samaity: is you hostcfgd up-to-date? If so then maybe there's a bug in hostcfgd?

Reply:
@jleveque I just checked again. It seems **hostcfgd** updates all the docker containers' hostname without a restart but not the SONiC switch's hostname itself. As per [#2924](https://github.com/Azure/sonic-buildimage/pull/2924) PR, **hostcfgd** keeps containers' hostname up-to-date. 

Am I missing something?

Reply:
@samaity: You're correct. It looks like the author of https://github.com/Azure/sonic-buildimage/pull/2924 was only concerned with updating the hostname inside the containers. Ultimately, this functionality should also be present for the host OS. This is the eventual goal for all SONiC components. However, this would involve eliminating hostname-config.service and moving that functionality into hostcfgd, which would require a good deal of testing. That is outside the scope of this PR. This solution is acceptable for now.